tests/test_jira_interaction.py

```diff
@@ -0,0 +1,258 @@
+        with self.assertLogs(level=WARNING) as cm:
+            dut.create_jira_issues(self.settings, None)
+        self.assertEqual(
+            cm.output,
+            ["WARNING:sphinx.mlx.traceability_exception:Configuration for automated ticket creation via Jira API is "
+             "missing mandatory values for keys ['api_endpoint']"]
+        )
+
+    def test_missing_username(self, *_):
+        self.settings.pop('username')
+        with self.assertLogs(level=WARNING) as cm:
+            dut.create_jira_issues(self.settings, None)
+        self.assertEqual(
+            cm.output,
+            ["WARNING:sphinx.mlx.traceability_exception:Configuration for automated ticket creation via Jira API is "
+             "missing mandatory values for keys ['username']"]
+        )
+
+    def test_missing_all_mandatory(self, *_):
+        mandatory_keys = ['api_endpoint', 'username', 'password', 'item_to_issue_regex', 'issue_type']
+        for key in mandatory_keys:
+            self.settings.pop(key)
+        with self.assertLogs(level=WARNING) as cm:
+            dut.create_jira_issues(self.settings, None)
+        self.assertEqual(
+            cm.output,
+            ["WARNING:sphinx.mlx.traceability_exception:Configuration for automated ticket creation via Jira API is "
+             "missing mandatory values for keys {}".format(mandatory_keys)]
+        )
+
+    def test_missing_all_optional_one_mandatory(self, *_):
+        mandatory_keys = ['components', 'project_key_prefix', 'project_key_regexp', 'default_project',
+                          'relationship_to_parent', 'warn_if_existent', 'password']
+        for key in mandatory_keys:
+            self.settings.pop(key)
+        with self.assertLogs(level=WARNING) as cm:
+            dut.create_jira_issues(self.settings, None)
+        self.assertEqual(
+            cm.output,
+            ["WARNING:sphinx.mlx.traceability_exception:Configuration for automated ticket creation via Jira API is "
+             "missing mandatory values for keys ['password']"]
+        )
+
+    def test_create_jira_issues_unique(self, jira):
+        jira_mock = jira.return_value
+        jira_mock.search_issues.return_value = []
+        with self.assertLogs(level=WARNING) as cm:
+            warning('Dummy log')
+            dut.create_jira_issues(self.settings, self.coll)
+
+        self.assertEqual(
+            cm.output,
+            ['WARNING:root:Dummy log']
+        )
+
+        self.assertEqual(jira.call_args,
+                         mock.call({'server': 'https://jira.atlassian.com/rest/api/latest/'},
+                                   basic_auth=('my_username', 'my_password')))
+        self.assertEqual(jira_mock.search_issues.call_args_list,
+                         [
+                             mock.call("project=MLX12345 and summary ~ 'MEETING-12345_2 Caption for action 1'"),
+                             mock.call("project=MLX12345 and summary ~ 'Caption for action 2'"),
```

Comment:
Interesting solution, I was mostly looking at `MagicMock` and `@patch` for such things.

Reply:
`jira_mock` is a `MagicMock` instance and `@patch` is used on the entire test suite. How else could I have done it?

Reply:
Hm, ok I missed that. Didn't know you can apply `@patch` to whole suite anyway. Cool.